driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -143,6 +144,47 @@ public static boolean serverVersionGreaterThan(final String versionString) {
         return getServerVersion().compareTo(new ServerVersion(getVersionList(versionString).subList(0, 3))) > 0;
     }
 
+    public static boolean canRunTests(final BsonArray runOn) {
+        boolean topologyFound = false;
+        ServerVersion serverVersion = getServerVersion();
+        for (BsonValue info : runOn) {
+            final BsonDocument document = info.asDocument();
+
+            if (document.containsKey("minServerVersion") && serverVersion.compareTo(getServerVersion("minServerVersion", document)) < 0) {
+                continue;
+            }
+            if (document.containsKey("maxServerVersion") && serverVersion.compareTo(getServerVersion("maxServerVersion", document)) < 0) {
+                continue;
+            }
+            if (document.containsKey("topology")) {
+                BsonArray topologyTypes = document.getArray("topology");
+                for (BsonValue type : topologyTypes) {
+                    String typeString = type.asString().getValue();
+                    if (typeString.equals("sharded")) {
+                        topologyFound = isSharded();
+                    } else if (typeString.equals("replicaset")) {
+                        topologyFound = isDiscoverableReplicaSet();
+                    } else if (typeString.equals("single")) {
+                        topologyFound = isStandalone();
+                    }
```

Comment:
Add an 

      else {
             throw  ...
      }

so we don't silently skip tests if for some reason the typeString doesn't match (like if someone camel-cases it).

Reply:
Good call - Done